stores/metadata.go

```diff
@@ -919,26 +919,38 @@ func (s *SQLStore) ContractSets(ctx context.Context) ([]string, error) {
 }
 
 func (s *SQLStore) ContractSizes(ctx context.Context) (map[types.FileContractID]api.ContractSize, error) {
-	rows := make([]struct {
+	type size struct {
 		Fcid     fileContractID `json:"fcid"`
 		Size     uint64         `json:"size"`
 		Prunable uint64         `json:"prunable"`
-	}, 0)
+	}
 
-	if err := s.db.
-		Raw(`
-SELECT fcid, MAX(c.size) as size, CASE WHEN MAX(c.size)>COUNT(cs.db_sector_id) * ? THEN MAX(c.size)-(COUNT(cs.db_sector_id) * ?) ELSE 0 END as prunable
-FROM contracts c
-LEFT JOIN contract_sectors cs ON cs.db_contract_id = c.id
-GROUP BY c.fcid
-	`, rhpv2.SectorSize, rhpv2.SectorSize).
-		Scan(&rows).
-		Error; err != nil {
+	var nullContracts []size
+	var dataContracts []size
+	if err := s.db.Transaction(func(tx *gorm.DB) error {
+		// use WHERE NOT EXISTS to catch contracts that would otherwise have been caught by a LEFT JOIN
```

Comment:
it would be bit more helpful to explain what this query does rather than what alternative we had for it. Someone who looks at this in a week probably won't remember what the left join was about. 

so e.g. "fetch all contracts that don't have any useful sectors in them anymore"